R snippets: Improve Mahalanobis Distance snippet

niederle opened the pull request:
The snippet, now returns in addition to the mahalanobis distance, the degrees of freedom, the p-value of each data point based on chi-squared distribution and a mahalanobis distance threshold which is calculate for a given p-value.

niederle committed:
R snippets: Improve Mahalanobis Distance snippet

The snippet, now returns in addition to the mahalanobis distance, the degrees of freedom, the p-value of each data point based on chi-squared distribution and a mahalanobis distance threshold which is calculate for a given p-value.

stoeter commented on knime-scripting-templates/R/snippet-templates.txt:
```diff
@@ -592,21 +592,23 @@ rOut <- data.frame(do.call("cbind", new.l))
 <!--1. Title and short description -->
 
-<h3 text="$$$TEMPLATE_NAME$$$" aligment="center" span="full"/>
+<h3 text="Mahalanobis Distance Calculator" aligment="center" span="full"/>
```
is there a reason to use a fixed text rather then getting the name of the template?

stoeter commented on knime-scripting-templates/R/snippet-templates.txt:
```diff
@@ -663,24 +671,27 @@ outInverted <- useInverted
+kIn['p-value'] <- pchisq(kIn$mahadist, df=nDegreesFreedom, lower.tail=FALSE)
+kIn['degrees of freedom'] <- nDegreesFreedom
+kIn['mahadist-threshold'] <- qchisq(p_thresh, nDegreesFreedom, lower.tail = FALSE)
+kIn$mahadistMode <- useInverted
```
No clue why there was the variable outInvert before. 
Conseuqently one should also remove the line 662/670 where outInvert is defined

niederle replied:
The old script kept the user setting of 'use inverted' in that variable to create an output column with that value. If the user had selected 'auto' the output column contained 'auto'. But I often thought it would be nice to know if 'auto' meant 'yes' or 'no' for the calculation in a particular case. I changed the script so that this column now will contain the actual decision if the inverted covariance matrix has been used. So there is no need to keep the setting anymore. I will remove it.

stoeter commented on knime-scripting-templates/R/snippet-templates.txt:
```diff
@@ -623,13 +625,13 @@ Alternatively, the message 'Error: Could not find atrribute: treatment' can be i
 </hbox>
 <gaprow height="1"/>
-
+<vector  var="p_thresh" label="Outlier p-value" size="1" vector-type="numeric" default-value="0.001"/>
```
I noticed a very strange result in the behaviour of RGG. If this volue 0.001 is changed it calculated the threshold correcly, but upon re-opening the snippet node configuration, the default value is back. 
I did not trouble shoot to the end, but it looks like it has to do with the vector-type. Play around with these two line of code...

<vector  var="p_thresh" label="Outlier p-value" size="1" vector-type="numeric" default-value="0.001"/>

<vector label="number of digits before/after decimal point" var="digitVector" size="1" default-value="3" vector-type="character"/>

niederle replied:
You are right. The numeric vector does not store the settings for some reason. I created an [RGG-issue](https://github.com/knime-mpicbg/rgg/issues/11)
As a workaround I will replace it with a non-numeric vector input and its value will be converted to a number.

stoeter reviewed (approved):
There are some comments and a "buggy" behaviour of RGG. I think it can be merged, but I think resolving the RGG thing would be nice. 
Thx! M.

niederle committed:
R snippet - mahalanobis distance

fix small issues like template name, workaround with non-numeric vector to store its setting and removal of an unused variable
